repo/content/content_manager_indexes.go

```diff
@@ -140,16 +165,6 @@ func (bm *Manager) addIndexBlobsToBuilder(ctx context.Context, bld packIndexBuil
 	}
 
 	_ = index.Iterate(AllIDs, func(i Info) error {
-		for _, dc := range opt.DropContents {
-			if dc == i.ID {
-				log(ctx).Debugf("dropping content %v", i.ID)
-				return nil
-			}
-		}
-		if i.Deleted && !opt.DropDeletedBefore.IsZero() && i.Timestamp().Before(opt.DropDeletedBefore) {
```

Comment:
It seems that this was assuming that there would only be a single entry for the "deleted" content, so it was enough to simply not include it in the resulting index.